parse/procedures/parser/visitor.go

```diff
@@ -144,24 +146,29 @@ func (p *proceduralLangVisitor) VisitExpr_boolean_literal(ctx *gen.Expr_boolean_
 		e.TypeCast = ctx.Type_cast().Accept(p).(*types.DataType)
 	}
 
+	e.Set(ctx)
+
 	return e
 }
 
 func (p *proceduralLangVisitor) VisitExpr_call(ctx *gen.Expr_callContext) any {
 	e := p.Visit(ctx.Call_expression())
 
+	var tc *types.DataType
 	if ctx.Type_cast() != nil {
-		tc := ctx.Type_cast().Accept(p).(*types.DataType)
-
-		switch v := e.(type) {
-		case *ExpressionCall:
-			v.TypeCast = tc
-		case *ExpressionForeignCall:
-			v.TypeCast = tc
-		default:
-			// should never happen
-			panic(fmt.Sprintf("invalid type cast for %T", e))
-		}
+		tc = ctx.Type_cast().Accept(p).(*types.DataType)
+	}
+
+	switch v := e.(type) {
+	case *ExpressionCall:
+		v.Set(ctx)
+		v.TypeCast = tc
+	case *ExpressionForeignCall:
+		v.Set(ctx)
+		v.TypeCast = tc
+	default:
+		// should never happen
+		panic(fmt.Sprintf("invalid type cast for %T", e))
```

Comment:
No, if we never expect them to happen I put a panic, since it is an internal error. Normally I would just return an error, but visitors cannot take errors, so I panic and catch it